packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/index.tsx

```diff
@@ -41,17 +34,26 @@ class Checkout extends PureComponent<Props> {
       )
       this.props.simpleBuyActions.createSBOrder(
         undefined,
-        formValues?.method?.type as SBPaymentMethodType['type']
+        this.props.method.type as SBPaymentMethodType['type']
       )
-    } else if (formValues && formValues.method) {
-      switch (formValues.method.type) {
+    } else if (formValues && !this.props.method) {
+      const fiatCurrency = this.props.fiatCurrency || 'USD'
+      this.props.simpleBuyActions.setStep({
+        step: 'PAYMENT_METHODS',
+        fiatCurrency,
+        pair: this.props.pair
+      })
+    } else if (formValues && this.props.method) {
+      switch (this.props.method.type) {
         case 'PAYMENT_CARD':
           this.props.simpleBuyActions.setStep({
             step: 'ADD_CARD'
           })
           break
         case 'USER_CARD':
-          this.props.simpleBuyActions.createSBOrder(formValues.method.id)
+          // TODO figure out id
+          // this.props.simpleBuyActions.createSBOrder(formValues.method.id)
+          this.props.simpleBuyActions.createSBOrder()
```

Comment:
combine cases
```
case 'USER_CARD'
case 'BANK_ACCOUNT'
  this.props.simpleBuyActions.createSBOrder()
  break
```

Reply:
yeah we have `TODO` there passing id as we had it before...